builder/buffer.go

```diff
@@ -13,25 +17,74 @@ var escapeCache sync.Map
 
 type escapeCacheKey struct {
 	value  string
-	prefix string
-	suffix string
+	quoter Quoter
 }
 
 // Buffer is used to build query string.
 type Buffer struct {
 	strings.Builder
+	Quoter              Quoter
+	ValueConverter      driver.ValueConverter
 	ArgumentPlaceholder string
 	ArgumentOrdinal     bool
-	EscapePrefix        string
-	EscapeSuffix        string
+	InlineValues        bool
+	BoolTrueValue       string
+	BoolFalseValue      string
 	valueCount          int
 	arguments           []interface{}
 }
 
 // WriteValue query placeholder and append value to argument.
 func (b *Buffer) WriteValue(value interface{}) {
-	b.WritePlaceholder()
-	b.arguments = append(b.arguments, value)
+	if !b.InlineValues {
+		b.WritePlaceholder()
+		b.arguments = append(b.arguments, value)
+		return
+	}
+
+	// Detect float bits to not lose precision after converting to float64
+	var floatBits = 64
+	if reflect.ValueOf(value).Kind() == reflect.Float32 {
+		floatBits = 32
+	}
+
+	if v, err := b.ValueConverter.ConvertValue(value); err != nil {
+		log.Printf("[WARN] unsupported inline value %v", value)
+	} else {
+		value = v
+	}
+
+	if value == nil {
+		b.WriteString("NULL")
+		return
+	}
+
+	switch v := value.(type) {
+	case string:
+		b.WriteString(b.Quoter.Value(v))
+		return
+	case []byte:
+		b.WriteString(b.Quoter.Value(string(v)))
+		return
+	}
+
+	rv := reflect.ValueOf(value)
+	switch rv.Kind() {
+	case reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64:
+		b.WriteString(strconv.FormatInt(rv.Int(), 10))
+		return
+	case reflect.Float32, reflect.Float64:
+		b.WriteString(strconv.FormatFloat(rv.Float(), 'g', -1, floatBits))
+		return
+	case reflect.Bool:
+		if rv.Bool() {
+			b.WriteString(b.BoolTrueValue)
+		} else {
+			b.WriteString(b.BoolFalseValue)
+		}
+		return
+	}
+	b.WriteString(fmt.Sprintf("%v", value))
```

Comment:
nice, looks like we can use this to fix default value:
https://github.com/go-rel/sql/blob/e49d9c759b8dc5f314ebef8b42cd27c5043648b6/builder/table.go#L162-L172

(maybe in different pr)